From 09413682ffef0d02382e8906f76e271e7f3e5757 Mon Sep 17 00:00:00 2001
From: Ghennadi Procopciuc <ghennadi.procopciuc@nxp.com>
Date: Thu, 12 Jan 2023 15:25:05 +0200
Subject: [PATCH 33/63] devicetree/bindings: stm: Add a property for clock
 prescaler

Clock prescaler should be used to have less count-up overflow
where this is needed.

Issue: ALB-9261
Upstream-Status: Pending 

Signed-off-by: Ghennadi Procopciuc <ghennadi.procopciuc@nxp.com>
Signed-off-by: Zhantao Tang <zhantao.tang@windriver.com>
---
 .../devicetree/bindings/clock/nxp-s32cc-stm-global.yaml    | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/Documentation/devicetree/bindings/clock/nxp-s32cc-stm-global.yaml b/Documentation/devicetree/bindings/clock/nxp-s32cc-stm-global.yaml
index 829221ef9ed0..03b4a164e739 100644
--- a/Documentation/devicetree/bindings/clock/nxp-s32cc-stm-global.yaml
+++ b/Documentation/devicetree/bindings/clock/nxp-s32cc-stm-global.yaml
@@ -36,6 +36,13 @@ properties:
   cpu:
     maxItems: 1
 
+  nxp,prescaler:
+        $ref: /schemas/types.yaml#/definitions/uint32
+        minimum: 1
+        maximum: 256
+        description:
+          Selects the module clock divide value for the prescaler, see timer's datasheet
+
   rs485-rts-active-low: true
   linux,rs485-enabled-at-boot-time: true
 
-- 
2.25.1

